plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -399,6 +406,18 @@ private static Optional<String> getComment(Document doc)
         return Optional.ofNullable(doc.getString(COMMENT_KEY));
     }
 
+    private boolean schemaExists(String schemaName)
+    {
+        Optional<String> remoteSchema = ImmutableList.copyOf(listDatabaseNames()).stream().filter(schemaName::equalsIgnoreCase).findFirst();
```

Comment:
We shouldn't iterate all database names if case insensitive name mapping is disabled.

Reply:
Does 'We shouldn't iterate all database names' mean that if case-insensitive name mapping is disabled(false), the result of `schemaExists` must always be true?

If implemented in this way, Schema is visible on Trino, but when executing CREATE/SELECT statements, the following MongoDB error is exposed to the user.

```
-- In mongo-client
> use CaseInsensitive;
switched to db CaseInsensitive
> use CaseInsensitiveDb;
switched to db CaseInsensitiveDb
> db.getCollection("test").insert({"a":"b"})
WriteResult({ "nInserted" : 1 })
> db.test.find({})
{ "_id" : ObjectId("63c6a185169d470a9f272353"), "a" : "b" }

-- in trino-cli
trino> show schemas in mongodb;
       Schema       
--------------------
 abc                
 admin              
 caseinsensitivedb  
 config             
 information_schema 
 local              
 test               
 testdatabase       
(8 rows)

Query 20230117_132900_00004_tviuz, FINISHED, 1 node
Splits: 19 total, 19 done (100.00%)
0.35 [8 rows, 110B] [22 rows/s, 315B/s]

trino> show tables in mongodb.caseinsensitivedb;
 Table 
-------
(0 rows)

Query 20230117_132917_00005_tviuz, FINISHED, 1 node
Splits: 19 total, 19 done (100.00%)
0.18 [0 rows, 0B] [0 rows/s, 0B/s]

trino> create table mongodb.caseinsensitivedb.testtable (name varchar);
Query 20230117_132932_00006_tviuz failed: Command failed with error 13297 (DatabaseDifferCase): 'db already exists with different case already have: [CaseInsensitiveDb] trying to create [caseinsensitivedb]' on server localhost:27017. The full response is {"ok": 0.0, "errmsg": "db already exists with different case already have: [CaseInsensitiveDb] trying to create [caseinsensitivedb]", "code": 13297, "codeName": "DatabaseDifferCase"}

trino> select * from mongodb.caseinsensitivedb.test;
Query 20230117_133109_00007_tviuz failed: line 1:15: Table 'mongodb.caseinsensitivedb.test' does not exist
select * from mongodb.caseinsensitivedb.test
```